docs/domain_modeling.md

```diff
@@ -566,6 +566,166 @@ Supplying this as a YAML file to the Chronicle build image as documented in
 [building chronicle](./building.md) will produce a well-typed API for your
 domain. The next step is then [recording provenance](./recording_provenance.md).
 
+## Chronicle Domain Definition Documentation Capabilities
+
+Chronicle's documentation capabilities provide users with a powerful tool for
+generating Rust code documentation and GraphQL schema documentation for their
+domains.
+
+Chronicle allows users to define a domain and generate documentation for that
+domain using documentation comments in the domain.yaml file. When Chronicle
+loads a valid domain.yaml file, it generates the domain on the basis of that
+file, and any `doc` field comments -- they are optional fields that can be
+left out -- found in the domain.yaml will be generated as Rust code
+documentation as well as documentation of the GraphQL schema for the given
+Chronicle domain.
```

Comment:
I'm finding this sentence a bit awkward. I'm not sure that the doc field comments are "generated" as other documentation, and the parenthetical bit about leaving optional parts out means a fair bit of parse context to reload to understand the last part of the sentence.